src/middlewared/middlewared/plugins/boot.py

```diff
@@ -34,13 +34,36 @@ async def get_state(self):
         # WebUI expects same data as `pool.pool_extend`
         return await self.middleware.call('pool.pool_normalize_info', BOOT_POOL_NAME)
 
+    @private
+    async def get_disks_cache(self):
+        """If boot pool disk cache hasn't been set (or cleared),
+        then set it.
+
+        NOTE: we cache this information since it doesn't change
+        very often and we have a ton of callers (especially on HA)
+        that need to determine this information. By caching this,
+        it reduces the amount of times we have to use our ProcessPool
+        """
+        global BOOT_POOL_DISKS
+        if BOOT_POOL_DISKS is None:
+            # Using an immutable object is very important since this is
+            # a globally cached value
+            BOOT_POOL_DISKS = tuple(await self.middleware.call('zfs.pool.get_disks', BOOT_POOL_NAME))
+        return BOOT_POOL_DISKS
```

Comment:
`return list(BOOT_POOL_DISKS)` here and you will never have to make a copy in callers.

Reply:
ahh yes, much cleaner that way!